### ISSUE TITLE
Bug of argument lr-scheduler

### ISSUE THREAD

**Issue by ghrua** [pytorch/fairseq#201]
The argument name in line  205 of options.py is `lr-scheduler`:
```
group.add_argument('--lr-scheduler', default='reduce_lr_on_plateau',
                   help='learning rate scheduler: {} (default: reduce_lr_on_plateau)'.format(
                       ', '.join(LR_SCHEDULER_REGISTRY.keys())))
```

While in line 87 of options.py it is `lr_scheduler`:
```
if hasattr(args, 'lr_scheduler'):
    LR_SCHEDULER_REGISTRY[args.lr_scheduler].add_args(parser)
```
So the `lr scheduler` couldn't registered correctly.

**Comment by myleott** [comment]
Python automatically converts the `-` into an underscore `_`, see: https://docs.python.org/3/library/argparse.html?highlight=argparse#dest.

Is registering new LR schedulers not working for you?

**Comment by ghrua** [comment]
@myleott Thanks for your help, I don't know this before. 

I tried to use `--lr_scheduler` in the command line which would be consistent with the argument name in line 87, and got `unrecognized arguments error`.

**Comment by myleott** [comment]
Ah, yes, you should use `--lr-scheduler` :)
